R/derive_param_tte.R

```diff
@@ -142,10 +159,261 @@
+#'   dataset_adsl = adsl,
+#'   by_vars = exprs(AEDECOD),
+#'   start_date = TRTSDT,
+#'   source_datasets = list(adsl = adsl, ae = ae_dup),
+#'   event_conditions = list(ttae),
+#'   censor_conditions = list(eos),
+#'   set_values_to = exprs(
+#'     PARAMCD = paste0("TTAE", as.numeric(as.factor(AEDECOD))),
+#'     PARAM = paste("Time to First", AEDECOD, "Adverse Event")
+#'   )
+#' )
+#'
+#' @info For investigating the issue, the dataset of the duplicate source records
+#'   can be obtained by calling `get_duplicates_dataset()`:
+#' @code
+#' get_duplicates_dataset()
+#'
+#' @info Common options to solve the issue in the `event_source()`/`censor_source()`
+#'   calls are:
+#' - Restricting the source records by specifying/updating the `filter` argument.
+#' - Specifying additional variables for `order`.
+#' - Setting `check_type = "none"` to ignore any duplicates.
+#'
+#' In this example it does not have significant impact which record is chosen as
+#' the dates are the same so the time to event derivation will be the same, but
+#' it does impact `SRCSEQ` in the output dataset, so here the second option is
+#' used.
+#' Note here how you can also define source objects from within the `derive_param_tte()`
+#' function call itself.
+#' @code
+#' derive_param_tte(
+#'   dataset_adsl = adsl,
+#'   by_vars = exprs(AEDECOD),
+#'   start_date = TRTSDT,
+#'   source_datasets = list(adsl = adsl, ae = ae_dup),
+#'   event_conditions = list(event_source(
+#'     dataset_name = "ae",
+#'     date = AESTDT,
+#'     set_values_to = exprs(
+#'       EVNTDESC = "AE",
+#'       SRCDOM = "AE",
+#'       SRCVAR = "AESTDTC",
+#'       SRCSEQ = AESEQ
+#'     ),
+#'     order = exprs(AESEQ)
+#'   )),
+#'   censor_conditions = list(eos),
+#'   set_values_to = exprs(
+#'     PARAMCD = paste0("TTAE", as.numeric(as.factor(AEDECOD))),
+#'     PARAM = paste("Time to First", AEDECOD, "Adverse Event")
+#'   )
+#' ) %>%
+#'   select(USUBJID, STARTDT, PARAMCD, PARAM, ADT, CNSR, SRCSEQ)
+#'
+#' @caption Filtering source records (`filter`)
+#' @info The first option could have been achieved using `filter`, for example
+#'   here only using serious adverse events.
+#' @code
+#' derive_param_tte(
+#'   dataset_adsl = adsl,
+#'   by_vars = exprs(AEDECOD),
+#'   start_date = TRTSDT,
+#'   source_datasets = list(adsl = adsl, ae = ae_dup),
+#'   event_conditions = list(event_source(
+#'     dataset_name = "ae",
+#'     filter = AESER == "Y",
+#'     date = AESTDT,
+#'     set_values_to = exprs(
+#'       EVNTDESC = "Serious AE",
+#'       SRCDOM = "AE",
+#'       SRCVAR = "AESTDTC",
+#'       SRCSEQ = AESEQ
+#'     )
+#'   )),
+#'   censor_conditions = list(eos),
+#'   set_values_to = exprs(
+#'     PARAMCD = paste0("TTSAE", as.numeric(as.factor(AEDECOD))),
+#'     PARAM = paste("Time to First Serious", AEDECOD, "Adverse Event")
+#'   )
+#' ) %>%
+#'   select(USUBJID, STARTDT, PARAMCD, PARAM, ADT, CNSR, SRCSEQ)
+#'
+#' @caption Using different censor values (`censor`)
```

Comment:
I wonder if the title should be changed. I think the main point is how to handle the case where the first of two censoring dates should be selected. If we state this in the title, it is easier to find.

Maybe we should also add an example (before this one) where two censorings are used and the first is selected. E.g., PFS.

Reply:
sure, will include this in the title. 

> Maybe we should also add an example (before this one) where two censorings are used and the first is selected. E.g., PFS.

not sure i understand this comment sorry, as the example given does already include 2 censoring conditions where the earliest occurring is selected. can you give more details please on the extra example you want to see added here?

Reply:
ok will look at adding one more example to make this default behaviour more clear. will use AE rather than PFS though as it's more universally accessible. once i have this one added, will request a re-review as have covered all the other comments now